client/src/error.rs

```diff
@@ -47,3 +54,16 @@ impl From<Box<state_machine::Error>> for Error {
 		ErrorKind::Execution(e).into()
 	}
 }
+
+impl From<state_machine::backend::Void> for Error {
+	fn from(_e: state_machine::backend::Void) -> Self {
+		unreachable!()
```

Comment:
`match e { }` better than the `unreachable!()` macro because it actually produces the LLVM unreachable intrinsic

Reply:
Isn't that a no-op?

Reply:
`match e { }` where e is an enum with no variants is impossible, so it forces the compiler to recognize that and it can lead to better code generation.